easybuild/easyblocks/n/namd.py

```diff
@@ -43,10 +44,21 @@ def extra_options():
             # see http://charm.cs.illinois.edu/manuals/html/charm++/A.html
             'charm_arch': [None, "Charm++ target architecture", MANDATORY],
             'charm_opts': ['--with-production', "Charm++ build options", CUSTOM],
-            'namd_basearch': ['Linux-x86_64', "NAMD base target architecture (compiler family is appended", CUSTOM],
             'namd_cfg_opts': ['', "NAMD configure options", CUSTOM],
             'runtest': [True, "Run NAMD test case after building", CUSTOM],
         })
+        arch = get_cpu_architecture()
+        if arch == X86_64:
+            basearch = 'Linux-x86_64'
+        elif arch == POWER:
+            basearch = 'Linux-POWER'
+
+        cuda = get_software_root('CUDA')
```

Comment:
This won't work as expected, since dependencies are not loaded yet at this point, this should be done in `prepare_step`...